### ISSUE TITLE
Cannot trigger actions when using sidekiq

### ISSUE THREAD

**Issue by luizkowalski** [websocket-rails/websocket-rails#287]
First of all, is this a bug?
I had this piece of code inside an worker

``` ruby
WebsocketRails["room_#{message.room.uid}"].trigger :new_message, message.as_json
```

this niver threw any error, but it doesn't work also. I just moved this to the controller, and my action is being trigger now. Maybe sidekiq and websocket-rails doesn't like each other. Is there a solution to this?

**Comment by confact** [comment]
I have the same problem. I use redis though, which should work, with channels. Still doesn't work.

**Comment by sebglazebrook** [comment]
+1

I'm seeing the same thing, using redis, synchronize => true and trying to access websocket_rails from Sidekiq.

**Comment by Samhane** [comment]
https://github.com/websocket-rails/websocket-rails/issues/295

I have the same problem
In debug, i see what resque use `RemoteConnection`, `Synchronization.publish event`, but no data in redis

**Comment by tranngocsam** [comment]
I have the same problem, but from Rails console, when I run this command, it returns an empty channels list

2.2.0 :002 > WebsocketRails.channel_manager.as_json
 => {"channels"=>{}}

**Comment by Ricardonacif** [comment]
Having the same problem as well. It's definately a bug on the syncronization. I'll take a deeper look.

**Comment by tranngocsam** [comment]
@Ricardonacif You're right, after looking up my app's synchronization settings, it works, thanks.

**Comment by e-tipalchuk** [comment]
Trying send message to user inside my controller in production
connection = WebsocketRails.users[user.id]
if @connection.connected?
    connection.send_message('new_message', {:message => 'Some message'})
end
and sometimes it works fine but sometimes WebsocketRails return MissingConnection

**Comment by Samhane** [comment]
WebsocketRails return MissingConnection if user not connected to server at this moment

**Comment by e-tipalchuk** [comment]
User is connected to server at this moment. The real situation is: I try to send 10 messages to connected user, 2-3 of them was received and other not send. Then I turn on logging for this method and i see that user disconnected and the connected and then disconnected and so on, but user is connected, and chrome dev tools shows me that everything is ok.

**Comment by scottbarrow** [comment]
Did anyone have a solution to this yet, also having and issue triggering from a sidekiq process

**Comment by scottbarrow** [comment]
I resolved this by enabling config.synchronize = true in config/initializers/websocket_rails.rb

**Comment by nicooga** [comment]
Wow, it seems there were a few people like me fighting with issue.
I spent almost 2 hours prying to this working, finally to realize that I just needed to put `config.synchronize = true` inf the config file.
I suggest making this even more obvious in the docs, or  even make this setting default to true.

**Comment by scottbarrow** [comment]
I just edited the wiki for clarification

**Comment by nicooga** [comment]
Awesome :+1:.

On Mon, Jan 4, 2016 at 8:06 PM, Scott Barrow <notifications@github.com>
wrote:

> I just edited the wiki for clarification
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/websocket-rails/websocket-rails/issues/287#issuecomment-168839792>
> .
>



-- 
--------------------------------------------------------------------------------------------
Nicolas Oga. Desarrollador de software, rubyista!
GitHub <http://nicooga.github.com>
LinkedIn <https://www.linkedin.com/profile/view?id=109143988>
Stackoverflow <http://stackoverflow.com/users/1740079/nicooga>
